packages/api/lambdas/sf-event-sqs-to-db-records/utils.js

```diff
@@ -125,10 +125,31 @@ const getMessageProviderCumulusId = async (cumulusMessage, knex) => {
   }
 };
 
+/**
+ * Get cumulus_id from insert query result run via knex.raw().
+ *
+ * knex.raw() does not return the same output structure as
+ * knex.insert().returning(), so we need to get the cumulus_id
+ * from the raw query result.
+ *
+ * @param {Object} result - Raw query result
+ * @returns {undefined|number}
+ *  Cumulus ID for the row. The value may also be undefined for
+ *  results of upsert queries with a WHERE clause where no rows
+ *  were updated by the query.
+ */
+const getCumulusIdFromRawInsertQueryResult = ({
+  rows = [],
+}) => {
+  const [row = {}] = rows;
```

Comment:
Does `const [row = {}] = rows` imply that we expect rows may look like `[{ "cumulus_id": 123 }]`?

Reply:
Indeed. But this piece of code is going anywhere entirely now that Knex has a new release with my changes, so I can use the Knex query builder with a `returning` statement and not have to parse a raw query result